cookbook/security/api_key_authentication.rst

```diff
@@ -67,7 +60,7 @@ value and then a User object is created::
         public function authenticateToken(TokenInterface $token, UserProviderInterface $userProvider, $providerKey)
         {
             $apiKey = $token->getCredentials();
-            $username = $this->userProvider->getUsernameForApiKey($apiKey);
+            $username = $userProvider->getUsernameForApiKey($apiKey);
```

Comment:
Don't we need to check first if the `ApiKeyAuthenticator` actually supports the given token? So, something like the following at the beginning of the method:

```php
if (!$this->supportsToken($token)) {
    return;
}
```